### PR TITLE
Added back-off and retry if rate limit is exceeded

### PR DESCRIPTION
**Pull request by zeus82**
I don't think this is the best implementation, but it's a start!

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @zeus82 has signed the CLA. Thanks for the pull request!

**Comment by coolboy** [comment]
And also please resolved the conflict

**Comment by coolboy** [comment]
@zeus82 conflict, please fix

**Comment by zeus82** [comment]
I will fix it this weekend.

File: Box.V2/Request/HttpRequestHandler.cs [hunk 1]
```diff
@@ -18,63 +18,80 @@ public class HttpRequestHandler : IRequestHandler
...
-
-                BoxResponse<T> boxResponse = new BoxResponse<T>();
-                boxResponse.Headers = response.Headers;
+                int attempts = 0;
+                BoxResponse<T> boxResponse = null;
 
-                // Translate the status codes that interest us 
-                boxResponse.StatusCode = response.StatusCode;
-                switch (response.StatusCode)
+                while (attempts < 10) // For now only 10 retries
```
Comment (zeus82) [inline Box.V2/Request/HttpRequestHandler.cs]: I agree... for now I'll put it as a constant, but it makes sense to make it configurable

**Comment by cburnette** [comment]
implemented in master already.

**Pull request closed without merging**